agent/dockerclient/mobyiface/interface.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Package mobyiface contains an interface for moby matching the
+// subset used by the agent
+package mobyiface
+
+import (
+	"context"
+	"io"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/network"
+	"github.com/docker/docker/api/types/events"
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/api/types/volume"
+)
+
+// Client is an interface specifying the subset of
+// github.com/moby/moby/tree/master/client that the agent uses.
+type Client interface {
+	Events(ctx context.Context, options types.EventsOptions) (<-chan events.Message, <-chan error)
```

Comment:
lint: i *think* you tried to align the methods with the existing ordering in the old interface. what do you think about grouping the methods by what the act on? Container*, Image*, Volume*.